### PR TITLE
[runtime env][bug fix] lock the minimum version of virtualenv to 20.0.24

### PR DESCRIPTION
**Pull request by Catch-Bull**
Signed-off-by: Catch-Bull <burglarralgrub@gmail.com>

## Why are these changes needed?

the test case will pass when virtualenv==20.0.24 but fails when virtualenv==20.0.23
test cmd: `pytest python/ray/tests/test_runtime_env.py::test_failed_job_env_no_hang -sv`

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py
Comment (architkulkarni) [inline python/setup.py:308]: Thanks! Can you also change it in `requirements/requirements.txt`?

Comment (Catch-Bull) [reply python/setup.py:308]: sure!

**Review by architkulkarni — APPROVED** [review]

**Comment by architkulkarni** [comment]
linkcheck and test_actor_failures flaky on master and unrelated

**Comment by SongGuyang** [comment]
@edoakes need your approval!

**Review by simon-mo — APPROVED** [review]
Thanks for the PR. @richardliaw @ericl code owners for `setup.py` PTAL

**Review by ericl — APPROVED** [review]

**Pull request merged**
